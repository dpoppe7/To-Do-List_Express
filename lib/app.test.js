const request = require('supertest');
const { describe, it, expect } = require('@jest/globals');

const app = require('./app');
const config = require('../../lib/config');
config.logLevel = 'silent';

const app = require('./app');

describe('the /hello endpoint', () => {
    it ('should respond with 200', () => 
        request(app)
            .get('/hello')
            .expect(200)
    );

    it ('should respond with plain text', () => 
        request(app)
            .get('/hello')
            .expect('Content-Type', /^text\/plain/)
    );

    it ('should match the snapshot', () => 
        request(app)
            .get('/hello')
            .expect(res => {
                expect(res.text).toMatchSnapshot()
            }
    ));
});
