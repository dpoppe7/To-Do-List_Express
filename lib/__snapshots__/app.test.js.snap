// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the /hello endpoint should match the snapshot 1`] = `
"Hello, world! 
You requested: /hello"
`;
